packages/discovery-provider/src/tasks/index_payment_router.py

```diff
@@ -123,6 +126,18 @@ class PurchaseMetadataDict(TypedDict):
     content_owner_id: int
 
 
+class RouteTransactionMemoType(str, enum.Enum):
+    purchase = "purchase"
+    recovery = "recovery"
+    unknown = "unknown"
+    none = "none"
```

Comment:
Makes sense. I think I was trying to make the code less conditional heavy and prone to lint errors by making sure the return value of the function always exists.
As it is, you already have to check against the type of the memo, then verify the nested metadata field exists. So it would mean something like `if (memo is not None and memo.type is Something and memo.metadata is not None): <treat it as a purchase>`

Reply:
Have updated this to drop the 'none' value